x-pack/plugins/cases/common/utils/markdown_plugins/utils.test.ts

```diff
@@ -18,5 +18,92 @@ describe('markdown utils', () => {
+        | col 1 is |  left-aligned | $1600 |
+        | col 2 is |    centered   |   $12 |
+        | col 3 is | right-aligned |    $1 |
+        "
+      `);
+    });
+
+    it('parses a timeline url', () => {
+      const timelineUrl =
+        '[asdasdasdasd](http://localhost:5601/moq/app/security/timelines?timeline=(id%3A%27e4362a60-f478-11eb-a4b0-ebefce184d8d%27%2CisOpen%3A!t))';
+
+      const parsedNodes = parseCommentString(timelineUrl);
+
+      expect(parsedNodes).toMatchInlineSnapshot(`
+        Object {
+          "children": Array [
+            Object {
+              "match": "[asdasdasdasd](http://localhost:5601/moq/app/security/timelines?timeline=(id%3A%27e4362a60-f478-11eb-a4b0-ebefce184d8d%27%2CisOpen%3A!t))",
+              "position": Position {
+                "end": Object {
+                  "column": 138,
+                  "line": 1,
+                  "offset": 137,
+                },
+                "indent": Array [],
+                "start": Object {
+                  "column": 1,
+                  "line": 1,
+                  "offset": 0,
+                },
+              },
+              "type": "timeline",
+            },
+          ],
+          "position": Object {
+            "end": Object {
+              "column": 138,
+              "line": 1,
+              "offset": 137,
+            },
+            "start": Object {
+              "column": 1,
+              "line": 1,
+              "offset": 0,
+            },
+          },
+          "type": "root",
+        }
+      `);
+    });
+
+    it('stringifies a timeline url', () => {
+      const timelineUrl =
+        '[asdasdasdasd](http://localhost:5601/moq/app/security/timelines?timeline=(id%3A%27e4362a60-f478-11eb-a4b0-ebefce184d8d%27%2CisOpen%3A!t))';
+
+      const parsedNodes = parseCommentString(timelineUrl);
+
+      expect(stringifyMarkdownComment(parsedNodes)).toEqual(timelineUrl);
+    });
+
+    it('parses a lens visualization', () => {
+      const lensVisualization =
+        '!{lens{"timeRange":{"from":"now-7d","to":"now","mode":"relative"},"editMode":false,"attributes":{"title":"TEst22","type":"lens","visualizationType":"lnsMetric","state":{"datasourceStates":{"indexpattern":{"layers":{"layer1":{"columnOrder":["col2"],"columns":{"col2":{"dataType":"number","isBucketed":false,"label":"Count of records","operationType":"count","scale":"ratio","sourceField":"Records"}}}}}},"visualization":{"layerId":"layer1","accessor":"col2"},"query":{"language":"kuery","query":""},"filters":[]},"references":[{"type":"index-pattern","id":"90943e30-9a47-11e8-b64d-95841ca0b247","name":"indexpattern-datasource-current-indexpattern"},{"type":"index-pattern","id":"90943e30-9a47-11e8-b64d-95841ca0b247","name":"indexpattern-datasource-layer-layer1"}]}}}';
+
+      const parsedNodes = parseCommentString(lensVisualization);
+      expect(parsedNodes.children[0].type).toEqual('lens');
+    });
+
+    it('stringifies a lens visualization', () => {
```

Comment:
This test will currently fail because we need to modify how the Plugins work for the `8.x` version of `remark-stringify`. I'll explain more later.